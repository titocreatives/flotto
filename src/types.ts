export type NavLink = {
  to: string;
  title: string | undefined;
};

export type NavLinks = NavLink[];

// manually because apollo codegen failed â†’ Apollo does not support anonymous operations

type draw = {
  date: string;
  numbers: string[];
  additionNumbers: string[];
};

export type drawData = {
  success: boolean;
  backendError: string;
  draws: draw[];
};

export type drawQueryResult = {
  draw: drawData;
};

export type Date = {
  month: string;
  day: string;
  year: string;
};

export type MonthTranslation = {
  Jan: string;
  Feb: string;
  Mar: string;
  Apr: string;
  May: string;
  Jun: string;
  Jul: string;
  Aug: string;
  Sep: string;
  Oct: string;
  Nov: string;
  Dec: string;
};

export enum LotteryTypes {
  eurojackpot = "eurojackpot",
  powerball = "powerball",
  megamillions = "megamillions"
}
