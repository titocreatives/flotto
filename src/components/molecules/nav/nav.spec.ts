import { mount, RouterLinkStub } from "@vue/test-utils";
import Nav from "./";
import { NavLinks } from "@/types";

const MOCK_LINKS: NavLinks = [
  {
    to: "/foo",
    title: "Foo"
  },
  {
    to: "/bar",
    title: "Bar"
  }
];

describe("Molecule Nav", () => {
  let cmp: any;

  beforeAll(() => {
    cmp = mount(Nav, {
      propsData: {
        links: MOCK_LINKS
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
  });

  it("matches snapshot", () => {
    expect(cmp.element).toMatchSnapshot();
  });

  it("passes Props to router-link", () => {
    const RouterLinks = cmp.findAll(RouterLinkStub);

    for (let i = 0; i < RouterLinks.length; i++) {
      expect(RouterLinks.at(i).props().to).toBe(MOCK_LINKS[i].to);
      expect(RouterLinks.at(i).text()).toEqual(MOCK_LINKS[i].title);
    }
  });
});
