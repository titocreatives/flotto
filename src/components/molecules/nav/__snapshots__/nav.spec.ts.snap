// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule Nav matches snapshot 1`] = `
<nav
  class="m nav"
>
  <a
    class="nav-item f-nav"
  >
    Foo
  </a>
  <a
    class="nav-item f-nav"
  >
    Bar
  </a>
</nav>
`;
